rmw_fastrtps_cpp/src/qos.hpp

```diff
@@ -39,7 +39,16 @@ bool
 get_datawriter_qos(
   const rmw_qos_profile_t & qos_policies,
   eprosima::fastrtps::PublisherAttributes & pattr);
-}
-// extern "C"
+
+RMW_LOCAL
+bool
+is_time_default(
+  const rmw_time_t & time);
+
+RMW_LOCAL
+bool
+is_valid_qos(
+  const rmw_qos_profile_t & qos_policies);
```

Comment:
C doesn't have pass by reference, so you shouldn't use it in an extern C.

Reply:
I see that existing functions do this, which is also wrong. It's basically a compiler feature, but we shouldn't rely on it unless we absolutely must:

https://stackoverflow.com/questions/1906000/c-by-reference-argument-and-c-linkage

Reply:
Likely these simply do not need to be inside of an extern C as they will not be called from C (they're not part of the rmw API for example).

Reply:
Agreed that the existing functions that do this are wrong. Also, if `extern "C"` is already used during the declaration of the function (such as in the header file), it is not needed again when defining the function (as in the .cpp file).

I have removed the use of `extern "C"` in this case.